src/stories/TextInput.tsx

```diff
@@ -0,0 +1,73 @@
+//import "./textInput.css";
+import { useState } from "react";
+import InfoIcon from "@mui/icons-material/Info";
+import KeyboardReturnIcon from "@mui/icons-material/KeyboardReturn";
+import TitleIcon from "@mui/icons-material/Title";
+interface TextInputProps {
+  placeholder: string;
+  isDisabled: boolean;
+  //isPassword: boolean;
+  //prefixIcon: React.ReactNode; //likely an icon in a div
+  //suffixIcon: React.ReactNode; //likely an icon in a div
+}
+
+const TextInput = ({
+  placeholder,
+  isDisabled,
+  //isPassword,
+  //prefixIcon,
+  //suffixIcon,
+}: TextInputProps) => {
+  const [value, setValue] = useState("");
+  const isValid = !/[0-9]/.test(value); //derived state
```

Comment:
As far as I can tell, this `isValid` function is arbitrary, particularly since the component is `TextInput`, not `NumberInput`. Even then, symbols such as `-` and `.` (both appearing only exactly once, and the former appearing 0 or 1 times at the start of the string) would be expected.

Reply:
Why are digits specifically not allowed?